### PR TITLE
Prepare release 1.50.0

### PR DESCRIPTION
**Pull request by albertteoh**
Updates changelog, release.md and jaeger-ui submodule for v1.50.0

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,13 +27,38 @@ next release (yyyy-mm-dd)
...
+* Deprecate grpc-storage-plugin sidecar model ([@yurishkuro](https://github.com/yurishkuro) in [#4744](https://github.com/jaegertracing/jaeger/pull/4744))
+* Upgrade query api v3 to official opentelemetry format ([@yurishkuro](https://github.com/yurishkuro) in [#4742](https://github.com/jaegertracing/jaeger/pull/4742))
+* [SPM] Deprecate support for spanmetrics processor naming convention ([@yurishkuro](https://github.com/yurishkuro) in [#4741](https://github.com/jaegertracing/jaeger/pull/4741))
+* Deprecate expvar metrics backend ([@yurishkuro](https://github.com/yurishkuro) in [#4740](https://github.com/jaegertracing/jaeger/pull/4740))
+* Fix flaky testgetroundtripper* tests ([@albertteoh](https://github.com/albertteoh) in [#4738](https://github.com/jaegertracing/jaeger/pull/4738))
+
 ### UI Changes
 
+[comment]: <> (TODO: update URL)
+* UI pinned to version [1.34.0](https://github.com/jaegertracing/jaeger-ui/blob/main/CHANGELOG.md#v1330-2023-08-06).
```
Comment (albertteoh) [inline CHANGELOG.md:60]: - [ ] Update URL and jaeger-ui submodule once https://github.com/jaegertracing/jaeger-ui/pull/1849 is merged.

**Review by yurishkuro — DISMISSED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (albertteoh) [reply CHANGELOG.md]: Done in: [f0a89e2](https://github.com/jaegertracing/jaeger/pull/4804/commits/f0a89e2fe008fcd25f7138449eaf2f78553ab19c)

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Strange that build started failing in this PR only, but here's a fix: #4807.

**Comment by albertteoh** [comment]
Checks passed with a false positive codecov check, @yurishkuro.

**Comment by yurishkuro** [comment]
My guess codecov is having a bad day

**Pull request merged**